evm/src/main/java/org/hyperledger/besu/evm/EVM.java

```diff
@@ -150,4 +159,33 @@ public Operation operationAtOffset(final Code code, final int offset) {
       return operation;
     }
   }
+
+  /**
+   * Determine whether a specified destination is a valid jump target.
+   *
+   * @param destination The destination we're checking for validity.
+   * @param code The code within which we are looking for the destination.
+   * @return Whether or not this location is a valid jump destination.
+   */
+  public boolean isValidJumpDestination(final UInt256 destination, final Code code) {
+    if (!destination.fitsInt()) return false;
+
+    final int jumpDestination = destination.intValue();
+    if (jumpDestination >= code.getSize()) return false;
+    long[] validJumpDestinations = code.getValidJumpDestinations();
+    if (validJumpDestinations == null || validJumpDestinations.length == 0) {
```

Comment:
If we are not putting the bits back on the `Code` object then this branch will always be visited.  We should either set it on the Code object or move the bit calculation code here and only look to the cache for the bits.

Reply:
The bits go back into the Code object [here](https://github.com/hyperledger/besu/blob/909df0cba9776f4713e3b74464cbb6094fe788b1/evm/src/main/java/org/hyperledger/besu/evm/Code.java#L115).

Reply:
I'm going to put some thought into this. 🤔 The Jump operations ask for this map from the EVM itself, not the Code.